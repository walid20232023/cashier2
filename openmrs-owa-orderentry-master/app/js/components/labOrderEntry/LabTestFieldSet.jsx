import React from 'react';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';
import { getConceptShortName } from '../../utils/helpers';
import '../../../css/grid.scss';

const LabTestFieldSet = (props) => {
  const {
    selectedTests, handleTestSelection, tests, labCategoryName, locale,
  } = props;
  const selectedTestIds = selectedTests.map(test => test.uuid);
  return (
    <fieldset className="fieldset">
      <legend>
        <FormattedMessage
          id="app.orders.tests"
          defaultMessage="Tests"
          description="Tests" />
      </legend>
      <div className="tests-box">
        {
          tests.length ?
            tests.map(test => (
              <button
                type="button"
                id="category-test-button"
                key={`${test.uuid}`}
                className={`lab-tests-btn ${(selectedTestIds.includes(test.uuid)) ? 'active' : ''}`}
                onClick={() => handleTestSelection(test, 'single')}
              >
                { getConceptShortName(test, locale) }
              </button>
            )) : <p>{labCategoryName} has no Standalone Tests</p>
        }
      </div>
    </fieldset>
  );
};

LabTestFieldSet.propTypes = {
  labCategoryName: PropTypes.string.isRequired,
  selectedTests: PropTypes.array.isRequired,
  handleTestSelection: PropTypes.func.isRequired,
  tests: PropTypes.array.isRequired,
};

export default LabTestFieldSet;
