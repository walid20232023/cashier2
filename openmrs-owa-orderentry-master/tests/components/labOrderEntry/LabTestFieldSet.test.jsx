import React from 'react';
import LabTestFieldSet from '../../../app/js/components/labOrderEntry/LabTestFieldSet';

let props;
let mountedComponent;
props = {
  handleTestSelection: jest.fn(),
  selectedTests: [
    { uuid: 'asampleduuid1-234', display: 'sampleA' },
    { uuid: 'defmpleduuid1-566', display: 'sampleB' },
  ],
  tests: [
    { uuid: 'asampleduuid1-234', display: 'sampleA' },
    { uuid: 'defmpleduuid1-566', display: 'sampleB' },
    { uuid: '5sampleduuid2-788', display: 'sampleC' }
  ]
};

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<LabTestFieldSet { ...props } />);
  }
  return mountedComponent;
};

describe('Component: LabTestFieldSet', () => {
  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should mount initially', () => {
    const component = getComponent();
    expect(component).toMatchSnapshot();
  });

  it('should support click for each button rendered', () => {
    const component = getComponent();
    const testButton = component.find('#category-test-button').at(0);
    testButton.simulate('click', {});
    expect(props.handleTestSelection).toBeCalled();
  });

  it('should add a class of `active` to the selected test', () => {
    const component = getComponent();
    expect(component.find('.lab-tests-btn.active').length).toEqual(2);
  })
});
