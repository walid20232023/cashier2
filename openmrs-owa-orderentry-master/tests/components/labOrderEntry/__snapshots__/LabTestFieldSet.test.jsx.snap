// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: LabTestFieldSet should mount initially 1`] = `
<fieldset
  className="fieldset"
>
  <legend>
    <FormattedMessage
      defaultMessage="Tests"
      description="Tests"
      id="app.orders.tests"
      values={Object {}}
    />
  </legend>
  <div
    className="tests-box"
  >
    <button
      className="lab-tests-btn active"
      id="category-test-button"
      key="asampleduuid1-234"
      onClick={[Function]}
      type="button"
    />
    <button
      className="lab-tests-btn active"
      id="category-test-button"
      key="defmpleduuid1-566"
      onClick={[Function]}
      type="button"
    />
    <button
      className="lab-tests-btn "
      id="category-test-button"
      key="5sampleduuid2-788"
      onClick={[Function]}
      type="button"
    />
  </div>
</fieldset>
`;
